src/Catty/Supporting Files/AppDelegate.swift

```diff
@@ -0,0 +1,121 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import UIKit
+
+func uncaughtExceptionHandler(exception: NSException) {
+    debugPrint("uncaught exception: ", exception.description)
+}
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    var window: UIWindow?
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        self.initNavigationBar()
+        ThemesHelper.changeAppearance()
+        let defaults = UserDefaults.standard
+        let appDefaults = NSDictionary.init(dictionary: ["lockiphone": "YES"]) as! [String: Any]
+        defaults.register(defaults: appDefaults)
+
+        self.setDefaultUserDefaults(defaults: defaults)
+        defaults.synchronize()
+        if ProcessInfo.processInfo.arguments.contains("UITests") {
+            UIApplication.shared.keyWindow?.layer.speed = 10.0
+        }
+
+        self.setupSiren()
+        self.setupFirebase()
+
+        return true
+    }
+
+    func applicationWillResignActive(_ application: UIApplication) {
+        let vc = self.window?.rootViewController as! UINavigationController
+
+        if (vc.topViewController?.isKind(of: ScenePresenterViewController.self))! {
+            let spvc = vc.topViewController as! ScenePresenterViewController
+            spvc.resumeAction()
```

Comment:
Please change all if-Conditions like 

`        if (vc.topViewController?.isKind(of: ScenePresenterViewController.self))! {
            let spvc = vc.topViewController as! ScenePresenterViewController
            spvc.resumeAction()
        }`

to

`        if let spvc = vc.topViewController as? ScenePresenterViewController {
            spvc.resumeAction()
        }`

It looks better and has the same behaviour.